### ISSUE TITLE
A GTK3 version of Nitrogen?

### ISSUE THREAD

**Issue by Hjdskes** [l3ib/nitrogen#27]
Hello,

I'm just curious whether or not you are considering/thinking about porting Nitrogen to GTK3. I changed Nitrogen's depends from gtkmm2/gtk2 to gtkmm3/gtk3 and on a first quick look, it doesn't look like it will require lots of changes to the code. However, I'm only starting to get coding in C and C++ I'm completely unfamiliar with. I tried porting some myself but alas, I couldn't get far.

Could you share your views on this? Perhaps you could offer Nitrogen in GTK3 in a new branch on GitHub, or make it possible through some configure flags.

I look forward to your replies. Regards,

Unia

**Comment by actionless** [comment]
how it's going? :3 

i've recently ported arandr to gtk3, and i would like to help with nitrogen as part of my goodbye-gtk2 initiative: https://github.com/themix-project/oomox-gtk-theme/issues/45

**Comment by Hjdskes** [comment]
@actionless I have something going in a project I no longer maintain. It might provide you a good start: https://github.com/Hjdskes/phosphorus

**Comment by actionless** [comment]
thanks for info! let's wait for author's reply; since i think it would be better to have it upstream instead of forking

**Comment by daf** [comment]
Hi all, I'm certainly open to Nitrogen getting to gtk3 (finally!)  @Hjdskes did some great work leading the charge with phosphorus, if stuff can be folded back in here that would be great.  I'm not sure I have the capacity to lead that myself yet, so I'd be happy if someone else were to get it started.

**Comment by actionless** [comment]
i didn't had big experience with c++ yet but i hope while porting existing code / merging with another port i think it should be not so hard for me

i'll create a PR as soon as i'll have something


@Hjdskes is your fork still mergeable back or there are some incompatible changes?

**Comment by Hjdskes** [comment]
Phosphorus is a from scratch project and written in C instead of Nitrogen's C++, so no, we can't fold anything back. However, you can copy ideas such as the plugin architecture.

**Comment by actionless** [comment]
motivation behind plugins is totally reasonable though i think both porting to gtk3 and to pear plugins could be too much

@Hjdskes do you have sorta todo list of phosphorus as of its current archived state?

**Comment by Hjdskes** [comment]
> @Hjdskes do you have sorta todo list of phosphorus as of its current archived state?

Not really, sorry. Last I remember running it it was functional in that it opens a window and loads a directory of wallpapers. There are plugins for GNOME and pure X11 desktops (window managers). I don't think there is support for multimonitor setups, but I might be wrong.

I suggest compiling it and giving it a go, that should give you an impression of the state it's in :)

**Comment by daf** [comment]
[Phosphorus' issues list](https://github.com/Hjdskes/phosphorus/issues) looks like a good roadmap of what was remaining.
